src/Shared/ReadOnlySpanExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Reflection;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Shared;
+
+internal static class ReadOnlySpanExtensions
+{
+    public static bool BinaryReadBoolean(this ReadOnlySpan<byte> span, ref int offset)
```

Comment:
```suggestion
    public static bool BinaryReadBooleanAndAdvance(this ReadOnlySpan<byte> span, ref int offset)
```
? Otherwise the side effect of `offset++` maybe astonishing.
Or add an xml-doc comment.